### PR TITLE
Expose Dex via Traefik URL

### PR DESCRIPTION
**Pull request by mhrabovcin**
Depends on:

✅ https://github.com/mesosphere/kubeaddons-extrasteps/pull/20
✅ https://github.com/mesosphere/charts/pull/161
⏳ https://github.com/mesosphere/charts/pull/166

### CODE REVIEW

**Comment by hectorj2f** [comment]
@mhrabovcin, this PR has been merged https://github.com/mesosphere/kubeaddons-extrasteps/pull/20, can we retag the images ?

File: templates/dex-k8s-authenticator.yaml [hunk 1]
```diff
@@ -57,5 +57,18 @@ spec:
           memory: 128Mi
       initContainers:
       - name: initialize-dka-config
-        image: mesosphere/kubeaddons-addon-initializer:v0.0.6
+        # TODO(mh): point to mesosphere/kubeaddons-addon-initializer:v0.0.9 once its tagged
+        # image: mesosphere/kubeaddons-addon-initializer:v0.0.9
+        image: mhrabovcin/kubeaddons-addon-initializer:latest
```
Comment (hectorj2f) [inline templates/dex-k8s-authenticator.yaml]: we need to re-tag this image before merging it

File: templates/dex.yaml [hunk 2]
```diff
@@ -23,8 +23,8 @@ spec:
       kubeaddons.mesosphere.io/provides: ingresscontroller
   chartReference:
     chart: dex
-    repo: https://mesosphere.github.io/charts/stable
-    version: 1.6.3
+    repo: https://timaa2k.github.io/charts
```
Comment (hectorj2f) [inline templates/dex.yaml]: we need to re-tag this image before merging it

File: templates/dex.yaml [hunk 3]
```diff
@@ -89,7 +89,8 @@ spec:
             - 'https://PUBLIC.URI/_oauth'
       initContainers:
       - name: initialize-dex
-        image: mesosphere/kubeaddons-addon-initializer:v0.0.7
+        # TODO(mh): point to mesosphere/kubeaddons-addon-initializer:v0.0.9 once its tagged
+        image: mhrabovcin/kubeaddons-addon-initializer:latest
```
Comment (hectorj2f) [inline templates/dex.yaml]: we need to re-tag this image before merging it

File: templates/kube-oidc-proxy.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    - name: none
+      enabled: true
+  requires:
+    matchLabels:
+      kubeaddons.mesosphere.io/provides: ingresscontroller
+  chartReference:
+    chart: kube-oidc-proxy
+    # TODO(mh): Point to main mesosphere repo once charts PR gets merged
+    # repo: https://mesosphere.github.io/charts/staging
+    repo: https://timaa2k.github.io/charts
```
Comment (hectorj2f) [inline templates/kube-oidc-proxy.yaml]: we need to change this repo before merging it

File: templates/kube-oidc-proxy.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        # updated by initcontainer when working with a remote dex
+        issuerUrl: https://dex-kubeaddons.kubeaddons.svc.cluster.local:8080/dex
+        usernameClaim: email
+        # placeholder : updated by initcontainer
+        caPEM: "placeholder"
+      initContainers:
+      - name: initialize-kube-oidc-proxy
+        # TODO(mh): point to mesosphere/kubeaddons-addon-initializer:v0.0.9 once its tagged
+        # image: mesosphere/kubeaddons-addon-initializer:v0.0.9
+        image: mhrabovcin/kubeaddons-addon-initializer:latest
```
Comment (hectorj2f) [inline templates/kube-oidc-proxy.yaml]: we need to re-tag this image before merging it

File: templates/traefik-forward-auth.yaml [hunk 6]
```diff
@@ -11,8 +11,10 @@ spec:
       kubeaddons.mesosphere.io/name: dex
   chartReference:
     chart: traefik-forward-auth
-    repo: https://mesosphere.github.io/charts/staging
-    version: 0.2.0
+    # TODO(mh): Point to main mesosphere repo once charts PR gets merged
+    # repo: https://mesosphere.github.io/charts/staging
+    repo: https://timaa2k.github.io/charts
```
Comment (hectorj2f) [inline templates/traefik-forward-auth.yaml]: we need to change this repo name before merging it

File: templates/traefik-forward-auth.yaml [hunk 7]
```diff
@@ -54,14 +57,19 @@ spec:
...
-      #    - name: "TRAEFIK_FORWARD_AUTH_NAMESPACE"
-      #      value: "kubeaddons"
-      #    - name: "TRAEFIK_FORWARD_AUTH_SECRET_NAME"
-      #      value: "ops-portal-credentials"
+      initContainers:
+      # initialize-traefik-forward-auth deploys credentials for use by the proxy
+      - name: initialize-traefik-forward-auth
+        # TODO(mh): point to mesosphere/kubeaddons-addon-initializer:v0.0.9 once its tagged
+        # image: mesosphere/kubeaddons-addon-initializer:v0.0.9
+        image: mhrabovcin/kubeaddons-addon-initializer:latest
```
Comment (hectorj2f) [inline templates/traefik-forward-auth.yaml]: we need to re-tag this image before merging it

File: templates/traefik.yaml [hunk 8]
```diff
@@ -57,3 +57,7 @@ spec:
         # separate issue.
         # See: https://jira.mesosphere.com/browse/DCOS-56033
         insecureSkipVerify: true
+
+      # TODO(mh): point to mesosphere/kubeaddons-addon-initializer:v0.0.9 once its tagged
+      # initCertJobImage: mesosphere/kubeaddons-addon-initializer:v0.0.9
+      initCertJobImage: mhrabovcin/kubeaddons-addon-initializer:latest
```
Comment (hectorj2f) [inline templates/traefik.yaml]: we need to re-tag this image before merging it

**Comment by hectorj2f** [comment]
@mhrabovcin This PR has some conflicts!

Also I assume we're waiting for the Konvoy to pass the tests. Am I right ?

**Comment by timaa2k** [comment]
This needs to change as we found problems with dex-k8s-auth not respecting HTTP_PROXY

File: templates/dex-k8s-authenticator.yaml [hunk 9]
```diff
@@ -23,8 +23,10 @@ spec:
       kubeaddons.mesosphere.io/name: dex
   chartReference:
     chart: dex-k8s-authenticator
-    repo: https://mesosphere.github.io/charts/staging
-    version: 1.1.6
+    # TODO(mh): Point to main mesosphere repo once charts PR gets merged
+    # repo: https://mesosphere.github.io/charts/staging
+    repo: https://timaa2k.github.io/charts
+    version: 1.1.7
```
Comment (jongiddy) [inline templates/dex-k8s-authenticator.yaml]: To allow dex-k8s-authenticator to use `HTTP_PROXY` settings, we'll be releasing an updated image, which will need to be tagged in this file.

File: templates/kube-oidc-proxy.yaml [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+      enabled: true
+    - name: docker
+      enabled: true
+    - name: none
+      enabled: true
+  requires:
+    matchLabels:
+      kubeaddons.mesosphere.io/provides: ingresscontroller
+  chartReference:
+    chart: kube-oidc-proxy
```
Comment (jieyu) [inline templates/kube-oidc-proxy.yaml]: this chart cannot be upgraded multiple times due to the use of hook resources
https://github.com/mesosphere/charts/blob/dev/staging/kube-oidc-proxy/templates/certificate.yaml#L8

Please fix that before releasing

**Pull request merged**